lib/index.js

```diff
@@ -44,20 +44,43 @@ export default function remarkTextr(options) {
   const settings = options || emptyOptions
   const plugins = settings.plugins || []
   const textrOptions = settings.options || {}
-  const promise = Promise.all(
-    plugins.map(async function (fn) {
-      if (typeof fn === 'string') {
-        // Assume plugin at default.
-        /** @type {{default: TextrPlugin}} */
-        const mod = await import(fn)
-        return mod.default
-      }
 
-      return fn
-    })
-  ).then(function (list) {
-    return textr(textrOptions).use(...list)
-  })
+  const noNeedPromise =
+    plugins.filter((plugin) => typeof plugin === 'string').length === 0
```

Comment:
Something along the lines of `needsPromise` and `Boolean(plugins.some((plugin) => typeof plugin === 'string'))` is easier to think about: it removes the “double negation”, and the `some` is faster as it exits on the first found item

Reply:
I implemented it with `plugins.some` in the beginning.  But some tests throw error, then I changed it into `filter`. Actually `Array.prototype.some` returns boolean already, so no need to use Boolean(). I will try and test it again.

Reply:
I tested it again, no problem, I used `Array.prototype.some`.

Reply:
Can you push again? I don’t see your changes yet!

Reply:
I was waiting the first unresolved, now I 've pushed.